repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/readme/ReadmePreProcessor.java

```diff
@@ -0,0 +1,109 @@
+import static org.apache.atlas.model.instance.AtlasEntity.Status.DELETED;
+import static org.apache.atlas.repository.Constants.QUALIFIED_NAME;
+import static org.apache.atlas.repository.Constants.STATE_PROPERTY_KEY;
+import static org.apache.atlas.repository.Constants.ASSET;
+
+public class ReadmePreProcessor implements PreProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(ReadmePreProcessor.class);
+
+    private final AtlasTypeRegistry    typeRegistry;
+    private final EntityGraphRetriever entityRetriever;
+    private final AtlasGraph           graph;
+    private final RestoreHandlerV1     restoreHandlerV1;
+
+    public ReadmePreProcessor(AtlasTypeRegistry typeRegistry, EntityGraphRetriever entityRetriever, AtlasGraph graph, RestoreHandlerV1 restoreHandlerV1) {
+        this.entityRetriever = entityRetriever;
+        this.typeRegistry = typeRegistry;
+        this.graph = graph;
+        this.restoreHandlerV1 = restoreHandlerV1;
+    }
+
+    @Override
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context,
+                                  EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("ReadmePreProcessor.processAttributes: pre processing {}, {}", entityStruct.getAttribute(QUALIFIED_NAME), operation);
+        }
+
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+
+        switch (operation) {
+            case CREATE:
+                processCreateReadme(entity, context);
+                break;
+            case UPDATE:
+                processUpdateReadme(entity, context.getVertex(entity.getGuid()));
+                break;
+        }
+    }
+
+    private void processCreateReadme(AtlasEntity entity, EntityMutationContext context) throws AtlasBaseException {
+        AtlasObjectId asset = (AtlasObjectId) entity.getRelationshipAttribute(ASSET);
+        if (asset != null) {
+            RequestContext requestContext = RequestContext.get();
+            AtlasPerfMetrics.MetricRecorder metricRecorder = requestContext.startMetricRecord("processCreateReadme");
+            String entityQualifiedName = createQualifiedName(asset);
+
+            entity.setAttribute(QUALIFIED_NAME, entityQualifiedName);
+
+            AtlasEntityType entityType = typeRegistry.getEntityTypeByName(entity.getTypeName());
+            AtlasVertex vertex = AtlasGraphUtilsV2.findByUniqueAttributes(graph, entityType, entity.getAttributes());
+            if (vertex != null) {
+                if (vertex.getProperty(STATE_PROPERTY_KEY, String.class).equals(DELETED.name())) {
+                    GraphTransactionInterceptor.addToVertexStateCache(vertex.getId(), AtlasEntity.Status.ACTIVE);
+                    restoreHandlerV1.restoreEntities(Collections.singletonList(vertex));
+                }
+                String internalGuidOfReadme = context.getDiscoveryContext().getReferencedGuids().get(asset.getGuid());
```

Comment:
What are we doing here? Asking out of curiosity

Reply:
Here, I'm getting readme's local guid. 
After making post request for readme, response have `guidAssignements` for readme which shows reference of `localGuid` to the `readme's guid`. Since we're updating the existing vertex so guid should remain same but reference in guidAssignments are newly created (`localGuid` -> `new guid of readme`) So to update guidAssignments I'm taking localGuid.